### PR TITLE
Release 2.2.0. PATCH support, improved timeout handling.

### PR DESCRIPTION
**Pull request by elland**

### CODE REVIEW

File: Teapot.podspec [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  spec.version = "2.2.0"
   spec.summary = "A light-weight URLSession wrapper for building simple API clients"
   spec.homepage = "https://github.com/bakkenbaeck/Teapot"
   spec.license = { type: 'MIT' }
   spec.authors = { "Igor Ranieri": "igor@elland.me" }
 
   spec.ios.deployment_target = '10.0'
   spec.requires_arc = true
-  spec.source = { git: 'https://github.com/bakkenbaeck/Teapot.git', tag: "#{spec.version}" }
+  spec.source = { git: 'https://github.com/elland/Teapot.git', tag: "#{spec.version}" }
```
Comment (yuliaveres) [inline Teapot.podspec:11]: this is not done yet 😏

File: Teapot/MockTeapot.swift [hunk 2]
```diff
@@ -91,7 +95,14 @@ open class MockTeapot: Teapot {
                 mockedError = TeapotError.invalidResponseStatus(self.statusCode.rawValue)
             }
 
-            let networkResult = NetworkResult(requestParameter, response!, mockedError)
+            let networkResult: NetworkResult
+
+            if let response = response, mockedError == nil {
+                networkResult = .success(requestParameter, response)
+            } else {
+                // we use ! as we've already stablished the error is non-nil before-hand.
```
Comment (yuliaveres) [inline Teapot/MockTeapot.swift]: established ?

File: Teapot/NetworkResult.swift [hunk 3]
```diff
@@ -18,15 +18,15 @@ import Foundation
...
+    case failure(RequestParameter?, HTTPURLResponse?, TeapotError)
 
-    public init(_ json: RequestParameter?, _ response: HTTPURLResponse, _ error: TeapotError? = nil) {
-        if let error = error {
-            self = .failure(json, response, error)
-        } else {
-            self = .success(json, response)
-        }
-    }
+//    public init(_ json: RequestParameter?, _ response: HTTPURLResponse?, _ error: TeapotError? = nil) {
```
Comment (yuliaveres) [inline Teapot/NetworkResult.swift]: please decide if we want to delete this code or uncomment

File: Teapot/MockTeapot.swift (hunk 2 above)
Comment (elland) [reply Teapot/MockTeapot.swift]: Mac keyboards……………

File: TeapotTests/DeliveryQueueTests.swift [hunk 4]
```diff
@@ -27,7 +31,7 @@ class DeliveryQueueTests: XCTestCase {
             expectation.fulfill()
         }
 
-        self.waitForExpectations(timeout: 20.0)
+        self.waitForExpectations(timeout: .defaultTimeout)
```
Comment (yuliaveres) [inline TeapotTests/DeliveryQueueTests.swift:34]: why this interval used to be so much longer before ?

File: Teapot.podspec (hunk 1 above)
Comment (yuliaveres) [reply Teapot.podspec:11]: 👀

File: TeapotTests/DeliveryQueueTests.swift (hunk 4 above)
Comment (elland) [reply TeapotTests/DeliveryQueueTests.swift:34]: I‘m not 100% sure anymore. I think it was just some default.

**Comment by elland** [comment]
No longer really needed, after Combine.

**Pull request closed without merging**